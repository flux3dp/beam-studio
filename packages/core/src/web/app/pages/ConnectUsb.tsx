/* eslint-disable @typescript-eslint/no-shadow */
import React, { useMemo } from 'react';
import { useLocation } from 'react-router-dom';
import classNames from 'classnames';

import useI18n from 'helpers/useI18n';

import { SupportUsbModels } from 'app/actions/beambox/constant';
import styles from './ConnectUsb.module.scss';

export default function ConnectUsb(): JSX.Element {
  const { initialize: t } = useI18n();
  const { search } = useLocation();
  const model = useMemo(
    () => new URLSearchParams(search).get('model'),
    [search]
  ) as SupportUsbModels;

  const renderInformation: Record<SupportUsbModels, { title: string; steps: Array<string> }> = {
    ado1: {
      title: 'Ador',
      steps: [
        t.connect_usb.turn_off_machine,
        t.connect_usb.tutorial1,
        t.connect_usb.turn_on_machine,
        t.connect_usb.wait_for_turning_on,
      ],
    },
    fhexa1: {
      title: 'HEXA',
      steps: [t.connect_usb.tutorial1, t.connect_usb.tutorial2],
    },
    fpm1: {
      title: 'Promark Series',
      steps: [t.connect_usb.tutorial1, t.connect_usb.connect_camera, t.connect_usb.tutorial2],
    },
    fbb2: {
      title: 'Beambox II',
      steps: [
        t.connect_usb.turn_off_machine,
        t.connect_usb.tutorial1,
        t.connect_usb.turn_on_machine,
        t.connect_usb.wait_for_turning_on,
      ],
    },
  };

  const handleNext = () => {
    const urlParams = new URLSearchParams({ model, usb: '1' });
    const queryString = urlParams.toString();

    window.location.hash = `#initialize/connect/connect-machine-ip?${queryString}`;
  };

  const renderStep = (model: SupportUsbModels) =>
    model ? (
      <div className={classNames(styles.contents, styles.tutorial)}>
        <div className={styles.subtitle}>{renderInformation[model].title}</div>
        <div className={styles.contents}>
          {renderInformation[model].steps.map((step, index) => (
            <div key={`usb-step-${index + 1}`}>
              {index + 1}. {step}
            </div>
          ))}
        </div>
      </div>
    ) : null;

  return (
    <div className={styles.container}>
      <div className={styles['top-bar']} />
      <div className={styles.btns}>
        <div className={styles.btn} onClick={() => window.history.back()}>
          {t.back}
        </div>
        <div className={classNames(styles.btn, styles.primary)} onClick={handleNext}>
          {t.next}
        </div>
      </div>
      <div className={styles.main}>
        <div className={styles.image}>
          <div className={classNames(styles.circle, styles.c1)} />
          <img src="img/init-panel/icon-usb-cable.svg" draggable="false" />
          <div className={classNames(styles.circle, styles.c2)} />
        </div>
        <div className={styles.text}>
          <div className={styles.title}>{t.connect_usb.title}</div>
          {renderStep(model)}
        </div>
      </div>
    </div>
  );
}
