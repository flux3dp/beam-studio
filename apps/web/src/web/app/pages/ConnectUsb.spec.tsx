import React from 'react';
import { render } from '@testing-library/react';

import ConnectUsb from './ConnectUsb';

jest.mock('helpers/useI18n', () => () => ({
  initialize: {
    connect_usb: {
      title: 'USB Connection',
      title_sub: ' (HEXA & Ador Only)',
      tutorial1: 'Connect the machine with your computer with USB cable.',
      tutorial2: 'Click Next.',
      turn_off_machine: 'Turn off the machine.',
      turn_on_machine: 'Turn on the machine.',
      wait_for_turning_on: 'Wait for the machine to turn on.',
    },
    next: 'Next',
    back: 'Back',
  },
}));
jest.mock('react-router-dom', () => {
  // Require the original module to not be mocked...
  const originalModule = jest.requireActual('react-router-dom');

  return {
    __esModule: true,
    ...originalModule,
    useLocation: jest.fn(() => ({
      pathname: '/initialize/connect/connect-usb',
      search: '?model=ado1',
      hash: '',
      state: null,
    })),
  };
});

describe('test ConnectUsb', () => {
  test('should render correctly', () => {
    const { container } = render(<ConnectUsb />);
    expect(container).toMatchSnapshot();
  });
});
