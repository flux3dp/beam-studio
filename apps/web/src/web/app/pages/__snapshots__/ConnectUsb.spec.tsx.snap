// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ConnectUsb should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="top-bar"
    />
    <div
      class="btns"
    >
      <div
        class="btn"
      >
        Back
      </div>
      <div
        class="btn primary"
      >
        Next
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="image"
      >
        <div
          class="circle c1"
        />
        <img
          draggable="false"
          src="img/init-panel/icon-usb-cable.svg"
        />
        <div
          class="circle c2"
        />
      </div>
      <div
        class="text"
      >
        <div
          class="title"
        >
          USB Connection
        </div>
        <div
          class="contents tutorial"
        >
          <div
            class="subtitle"
          >
            Ador
          </div>
          <div
            class="contents"
          >
            <div>
              1
              . 
              Turn off the machine.
            </div>
            <div>
              2
              . 
              Connect the machine with your computer with USB cable.
            </div>
            <div>
              3
              . 
              Turn on the machine.
            </div>
            <div>
              4
              . 
              Wait for the machine to turn on.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
